crates/driver/src/infra/solver/dto/auction.rs

```diff
@@ -16,6 +18,17 @@ impl Auction {
         timeout: competition::SolverTimeout,
         now: infra::time::Now,
     ) -> Self {
+        let liquidity_tokens = liquidity
+            .iter()
+            .flat_map(|l| match &l.kind {
+                liquidity::Kind::UniswapV2(pool) => {
+                    pool.reserves.iter().map(|r| r.token).collect_vec()
+                }
+                liquidity::Kind::UniswapV3(pool) => vec![pool.tokens.get().0, pool.tokens.get().1],
+                _ => vec![],
+            })
+            .map(|t| (t.0.into(), Token::default()));
```

Comment:
Quasimodo's `/solve` requests will crash if it doesn't find all the liquidity tokens in the list with all the token meta data.. This showed one thing that should probably be redesigned:
Currently we initialize the list of token meta data when we don't yet know which liquidity we'll include in the settlement which means we could miss some.

Reply:
Do you mind creating an issue for this?